packages/language-server/src/hover.ts

```diff
@@ -0,0 +1,255 @@
+    }
+  }
+
+  isInsideRange(name: XMLToken | undefined): boolean {
+    if (
+      name !== undefined &&
+      this.chosenOffset !== undefined &&
+      name.startOffset <= this.chosenOffset &&
+      name.endOffset >= this.chosenOffset
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+}
+
+export function getHoverContext(
+  model: UI5SemanticModel,
+  textDocumentPosition: TextDocumentPositionParams,
+  document: TextDocument
+): Hover | undefined {
+  const documentText = document.getText();
+  const { cst, tokenVector } = parse(documentText);
+  const ast = buildAst(cst as DocumentCstNode, tokenVector);
+  const offset = document.offsetAt(textDocumentPosition.position);
+  const ui5Node = getUI5Node(ast, offset, model);
+  if (ui5Node != undefined) {
+    return transformToLspHover(ui5Node, model);
+  }
+
+  return undefined;
+}
+
+function getUI5Node(
+  ast: XMLDocument,
+  offset: number,
+  model: UI5SemanticModel
+): BaseUI5Node | undefined {
+  const visitor = new HoverContextVisitor(offset);
+  accept(ast, visitor);
+  if (visitor.hoverContext?.kind === undefined) {
+    return undefined;
+  }
+  switch (visitor.hoverContext?.kind) {
+    case "XMLElementHover":
+      return getUI5NodeByElement(visitor.hoverContext.astNode, model);
+    case "XMLKeyAttributeHover":
+      return getUI5NodeByKey(visitor.hoverContext.astNode, model);
+    case "XMLValueAttributeHover":
+      return getUI5NodeByValue(visitor.hoverContext.astNode, model);
+    case undefined:
+      return undefined;
+  }
+}
+
+function transformToLspHover(
+  ui5Node: BaseUI5Node,
+  model: UI5SemanticModel
+): Hover | undefined {
+  const hoverItem: Hover = {
+    contents: getNodeDocumentation(ui5Node, model),
+  };
+  return hoverItem;
+}
+
+function getUI5NodeByElement(
+  astNode: XMLElement,
+  model: UI5SemanticModel
+): UI5Class | UI5Aggregation | undefined {
+  const ui5Class = find(
+    model.classes,
+    (ui5class) => xmlToFQN(astNode) === ui5NodeToFQN(ui5class)
+  );
+
+  if (astNode.parent.type === "XMLDocument" || ui5Class != undefined) {
+    return getClassByElement(astNode, model);
+  } else {
+    const parentElementClass = getClassByElement(astNode.parent, model);
+    return findAggragationByName(parentElementClass, astNode.name);
+  }
+}
+
+function getUI5NodeByKey(
+  astNode: XMLAttribute,
+  model: UI5SemanticModel
+): UI5Prop | UI5Event | UI5Association | undefined {
+  const parentElementClass = getClassByElement(astNode.parent, model);
+  return findUI5ClassMemberByName(parentElementClass, astNode.key);
+}
+
+function getUI5NodeByValue(
+  astNode: XMLAttribute,
+  model: UI5SemanticModel
+): UI5Namespace | UI5EnumValue | undefined {
+  if (astNode.key != null && isXMLNamespaceKey(astNode.key)) {
+    const ui5Namespace = find(
+      model.namespaces,
+      (namespace) => astNode.value === ui5NodeToFQN(namespace)
+    );
+    return ui5Namespace;
+  } else {
```

Comment:
Open question: 

Should we assume it must be an enum and try to resolve as an enum
or should we first check if its an enum and only then handle the enum case?